01-2024-schedule/course-schedule/week-4.md

```diff
@@ -7,3 +7,66 @@ description: Second-Layer Applications
 
 Additionally, we'll delve into topics like path finding and how payments are delivered in the Lightning Network. Plus, we'll try to grasp the philosophy behind Bitcoin.
+
+## Prerequisites
+
+* LND should be installed and ready
+* Developer environment from week 1 should be available. Having [Polar](https://lightningpolar.com) installed and functional is essential! It requires [Docker](https://www.docker.com/products/docker-desktop/) to be installed.
+
+## Goals
+
+* Understand how the lightning network operates
+  * Why it is thought of as a bitcoin "layer 2"
+  * Where the trust in the system lies
+  * What security assumptions it makes
+  * How the various "layers" of the network fit together
+* Lightning Nodes and Wallets
+* Understand the architecture of Lightning Network
+* Be familiar with sending and receiving lightning payments
+* Understand Payment Channels
+  * Constructing a Payment Channel
+  * Sending Payment accross the Channel
+  * The Commitment Transaction
+  * Advancing the Channel State
+  * Closing the Channel
+* Understand Routing on a network of Payment Channels
+  * Routing a Payment
+  * Routing vs pathfinding
+  * Creating a network of Payment Channels
+  * Fairness Protocol
+  * Hash Time-Locked Contracts (HTLCs)
+* Understand Channel Operation and Payment Forwarding
+  * Managing HTLCs
+  * Committing HTLCs to channel state
+  * Channel balance
+* Understand Path Finding and Payment Delivery
+  * Finding Candidate Paths
+  * Payment Delivery
+  * Multiparts Payment (MPP)
+* Understand Lightning Payments Requests
+  * Payment Invoices
+  * Payment Requests vs Bitcoin Addresses
+  * Payment Requests Serialization and Interpretation
+
+## Schedule
+
+
+| Topic        | Goals               | Resources                                                         |
+| ------------ | --------------------|------------------------------------------------------------------ |
+| Lightning Network Basics     | Understand: <br/> Why Lightning is called Layer 2 <br/> Trust in Decentralised Network <br/> Security Assumptions <br/> The Lightning Network components |[Chapter 1: Introduction](https://lnbook.256k1.dev/#intro_what_is_the_lightning_network) <br/> [How Network Layers Fit Together](https://youtu.be/krux2v0jt4E?list=PLpLH33TRghT17_U3as2P3vHfAGL8pSOOY) |
+| Lightning Nodes and Wallets   | Understand:  Lightning Nodes <br/> Wallets |[Lightning Nodes](https://lnbook.256k1.dev/#_lightning_nodes) <br/> [Lightning Wallets](https://lnbook.256k1.dev/#_lightning_wallets)                                                   |
+| Lightning Network Architecture    | Understand the various components of the Lightning Network |[Chapter 6: Lightning Network Architecture](https://github.com/lnbook/lnbook/blob/develop/06_lightning_architecture.asciidoc#lightning-network-architecture)                                                                  |
```

Comment:
Same here. This should come before Lightning Nodes and Wallets on line 57

Reply:
Done.